test/TritonGPU/amd/amd-instruction-sched.mlir

```diff
@@ -0,0 +1,154 @@
+
+#blocked0_ex1 = #triton_gpu.blocked<{sizePerThread = [1, 8], threadsPerWarp = [8, 8], warpsPerCTA = [4, 1], order = [1, 0]}>
+#blocked1_ex1 = #triton_gpu.blocked<{sizePerThread = [8, 1], threadsPerWarp = [8, 8], warpsPerCTA = [1, 4], order = [0, 1]}>
+#blocked2_ex1 = #triton_gpu.blocked<{sizePerThread = [1, 8], threadsPerWarp = [4, 16], warpsPerCTA = [4, 1], order = [1, 0]}>
+#shared0_ex1 = #triton_gpu.shared<{vec = 8, perPhase = 1, maxPhase = 8, order = [1, 0], hasLeadingOffset = false}>
+#shared1_ex1 = #triton_gpu.shared<{vec = 8, perPhase = 1, maxPhase = 8, order = [0, 1], hasLeadingOffset = false}>
+#mma0_ex1 = #triton_gpu.amd_mfma<{versionMajor = 3, versionMinor = 0, warpsPerCTA = [2, 2], instrShape = [32, 32], isTransposed = false}>
+#dot0_ex1 = #triton_gpu.dot_op<{opIdx = 0, parent = #mma0_ex1, kWidth = 8}>
+#dot1_ex1 = #triton_gpu.dot_op<{opIdx = 1, parent = #mma0_ex1, kWidth = 8}>
+
+module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 4 : i32, "triton_gpu.threads-per-warp" = 64 : i32} {
+  // INSERT-LABEL: test_dot1
+  // INSTR_COUNT: test_dot1
+  // LOWER_IGLP0-LABEL: test_dot1
+  tt.func @test_dot1(
+    %arg0: tensor<32x32xf16, #triton_gpu.dot_op<{opIdx = 0, parent = #mma0_ex0, kWidth = 16}>>,
+    %arg1: tensor<32x32xf16, #triton_gpu.dot_op<{opIdx = 1, parent = #mma0_ex0, kWidth = 16}>>,
+    %arg2: tensor<32x32xf16, #mma0_ex0>) {
+
+    %c0_i32 = arith.constant 0 : i32
+    %c1_i32 = arith.constant 1 : i32
+    %c64_i32 = arith.constant 1 : i32
+
+    scf.for %arg11 = %c0_i32 to %c64_i32 step %c1_i32 iter_args() -> () : i32 {
+      // INSERT: amdgpu.instruction_sched_hint
+      // INSTR_COUNT: amdgpu.instruction_sched_hint
+      // INSTR_COUNT-SAME: numMMAs = #amdgpu.InstCounter<4, tensor<32x32x8xf16>
+      // LOWER_IGLP0: %346 = llvm.mlir.constant(0 : i32) : i32
+      // LOWER_IGLP0: llvm.call_intrinsic "llvm.amdgcn.iglp.opt"(%346) : (i32) -> ()
+      %0 = tt.dot %arg0, %arg1, %arg2, inputPrecision = ieee : tensor<32x32xf16, #triton_gpu.dot_op<{opIdx = 0, parent = #mma0_ex0, kWidth = 16}>> * tensor<32x32xf16, #triton_gpu.dot_op<{opIdx = 1, parent = #mma0_ex0, kWidth = 16}>> -> tensor<32x32xf16, #mma0_ex0>
+      scf.yield
+    }
+    tt.return
+  }
+
+  // INSTR_COUNT-LABEL: @test_dot2
+  tt.func public @test_dot2(
+    %arg0: !tt.ptr<f16> {tt.divisibility = 16 : i32},
+    %arg1: !tt.ptr<f16> {tt.divisibility = 16 : i32}
+    ) attributes {noinline = false} {
+
+    %cst = arith.constant dense<64> : tensor<256x64xi32, #blocked0_ex1>
+    %cst_0 = arith.constant dense<64> : tensor<64x128xi32, #blocked1_ex1>
+    %c0_i32 = arith.constant 0 : i32
+    %c1_i32 = arith.constant 1 : i32
+    %c64_i32 = arith.constant 64 : i32
+    %c63_i32 = arith.constant 63 : i32
+    %c128_i32 = arith.constant 128 : i32
+    %c256_i32 = arith.constant 256 : i32
+
+    %19 = tt.make_range {end = 256 : i32, start = 0 : i32} : tensor<256xi32, #triton_gpu.slice<{dim = 1, parent = #blocked0_ex1}>>
```

Comment:
Are all of these essential to writing the lit test? what about simplifying/deleting parts or use function parameters?

Reply:
My initial idea was to put everything into the function parameters to shorten the test MLIR code but I bumped into a problem during lowering to the LLVM dialect. The problem was related to some missing information in the `ModuleAxisInfoAnalysis` analysis. The solution that I found was writing the `for-loop` prologue (which can see in the test) which does pointer-arithmetic.